src/app/modules/user-offers-page/user-sale-offers-page/user-sale-offers-page.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, inject, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { CommonModule } from '@angular/common';
+import { AppUserService } from '../../../core/services/app-user.service';
+import { MySaleOffersComponent } from '../../my-offers/my-sale-offers/my-sale-offers.component';
+
+@Component({
+  selector: 'app-user-sale-offers-page',
+  imports: [CommonModule, MySaleOffersComponent],
+  templateUrl: './user-sale-offers-page.component.html',
+  styleUrl: './user-sale-offers-page.component.scss'
+})
+export class UserSaleOffersPageComponent implements OnInit {
+  private route = inject(ActivatedRoute);
+  private appUserService = inject(AppUserService);
+
+  userId: string | null = null;
+  userName: string | null = null;
+
+ngOnInit(): void {
+  const userIdFromRoute = this.route.snapshot.paramMap.get('id');
+
+  if (userIdFromRoute) {
+    this.userId = userIdFromRoute;
+
+    this.appUserService.getUser(this.userId).subscribe(user => {
```

Comment:
bo zmianie z BehaviorSubject na signal też będzie łatwiej i nie potrzebujesz przypisywać już np username do jakeijś zmiennej, po prostu jak będziesz miał coś w tym AppUserService w currentUser to będziesz mógł zrobić this.appUserService.currentUser.displayName